src/Symfony/Component/Mailer/Command/MailerDebugCommand.php

```diff
@@ -0,0 +1,129 @@
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+
+/**
+ * Helps making sure your mailer provider is operational.
+ *
+ * @author Guillaume MOREL <me@gmorel.io>
+ *
+ * @final
+ */
+class MailerDebugCommand extends Command
+{
+    /** {@inheritdoc} */
+    protected static $defaultName = 'debug:mailer:email:send';
+
+    /** @var MailerInterface */
+    private $mailer;
+
+    /** @var SymfonyStyle */
+    private $io;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(MailerInterface $mailer)
+    {
+        parent::__construct();
+
+        $this->mailer = $mailer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this->setDescription('Send simple email message')
+            ->setDefinition([
+                new InputOption('from', null, InputOption::VALUE_REQUIRED, 'The from email of the message'),
+                new InputOption('to', null, InputOption::VALUE_REQUIRED, 'The to email of the message'),
+                new InputOption('subject', null, InputOption::VALUE_REQUIRED, 'The subject of the message'),
+                new InputOption('body', null, InputOption::VALUE_REQUIRED, 'The body of the message'),
+            ])
+            ->setHelp(
+                <<<EOF
+The <info>%command.name%</info> command creates and sends a simple email message.
+Usage:
+- <info>php %command.full_name%</info> will trigger the interactive mode
+- <info>php %command.full_name% --from=a@symfony.com --to=b@symfony.com --subject=Test --body=body</info>
+EOF
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws TransportExceptionInterface
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $email = $this->createEmail($input);
+
+        try {
+            $this->mailer->send($email);
+        } catch (TransportExceptionInterface $e) {
+            $this->io->error('Fail to send email.');
+            throw $e;
+        }
+
+        $this->io->success(
+            sprintf(
+                'Email was successfully sent to "%s".',
+                (string) $input->getOption('to')
+            )
+        );
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        foreach ($input->getOptions() as $option => $value) {
+            if (null === $value) {
+                $input->setOption($option, $this->io->ask(sprintf('%s', ucfirst($option))));
+            }
+        }
+    }
+
+    private function createEmail(InputInterface $input): Email
+    {
+        return (new Email())
+            ->from($input->getOption('from'))
```

Comment:
This still does not guarantee that you have a value for the options, if the command was used in a non-interactive mode (which does not call `interact`). Options are optional by design.

Also note that `interact` is about interacting *before* the full validation of arguments, so it is able to fill missing arguments.

Reply:
interactive feature removed